expo/eas-build
# [build-tools] Remove classic update releaseChannel defaulting in expo-updates >= 0.19.0 SDK 50

# Why

As of SDK 50, classic updates is no longer supported. So builds should no longer configure a classic release channel and should instead configure nothing even when "releaseChannel" is set in eas.json for expo-updates.

Partially fixes ENG-11121.

# How

So the logic is the following when `expo-updates` version >= 0.19.0 (SDK 50 first alpha publish):
- configure for EAS update (or any modern updates) if updates.channel is set in the job (same behavior as before)
- otherwise, do nothing

Previously, and for `expo-updates` version < 0.19.0, the behavior should remain unchanged:
- configure for EAS update (or any modern updates) if updates.channel is set in the job (same behavior as before)
- configure for classic updates, set releaseChannel if specified

When no `releaseChannel` is specified in eas.json, the library itself used to default to "default". It no longer does.

That being said, I still need to figure out why the build job itself is getting "releaseChannel : default" in the build metadata during a normal `eas build`.

# Test Plan

Run tests.

Manual test:

Check SDK 49, ensure classic updates is still configured by default when expo-updates is installed but EAS update is not configured:
1. `yarn create expo-app try-eas-upgrades-3 --template blank@sdk-49`
2. `npx expo install expo-updates`
3. `neas init`
4. add `"releaseChannel": "testchannel"` to each build profile in eas.json
5. `~/expo/run-with-local-eas-build.sh build --local`
6. go to result dir, inspect AndroidManifest.xml in build intermediates, see that release channel is configured

Check SDK 50, ensure classic updates is not longer configured by default when expo-updates is installed but EAS update is not configured:
1. `yarn create expo-app try-eas-upgrades-4 --template blank`
2. `npx expo install expo-updates`
3. `neas init`
4. add `"releaseChannel": "testchannel"` to each build profile in eas.json
5. `~/expo/run-with-local-eas-build.sh build --local`
6. go to result dir, inspect AndroidManifest.xml in build intermediates, see that release channel is not configured
